src/Playwright/Core/Waiter.cs

```diff
@@ -117,6 +117,18 @@ internal void RejectImmediately(Exception exception)
         _immediateError = exception;
     }
 
+    internal void RejectOnCancellation(CancellationToken cancellationToken)
+    {
+        if (!cancellationToken.CanBeCanceled)
+        {
+            return;
+        }
+
```

Comment:
Just an idea: maybe it would be good to call `cancellationToken.ThrowIfCancellationRequested()` (here or maybe higher up in the call stack) so that if it is already cancelled we don't do anything.

Reply:
Yes, that was my understanding of your comment as well, alas I have added the throw in the wrong place, fixed now :)